### PR TITLE
fix: await remote even when instance is present in memory cache

### PR DESCRIPTION
**Pull request by tshedor**

### CODE REVIEW

File: packages/brick_offline_first/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,9 @@
 ## Unreleased
 
+## 3.1.0
+
 * Apply standardized lints
+* Respect `awaitRemote` policy for `OfflineFirstRepository#get` - fixes a bug where an instance in memory cache would early-return before hitting remote despite the requested `await` or `alwaysHydrate` policy
```
Comment (mateominato) [inline packages/brick_offline_first/CHANGELOG.md:6]: i feel like i can point to this for bugs we looked at years ago in GB register

File: packages/brick_offline_first/lib/src/offline_first_repository.dart [hunk 2]
```diff
@@ -175,18 +175,24 @@ abstract class OfflineFirstRepository<RepositoryModel extends OfflineFirstModel>
...
+    final hydrateUnexisting = policy == OfflineFirstGetPolicy.awaitRemoteWhenNoneExist;
+    final alwaysHydrate = policy == OfflineFirstGetPolicy.alwaysHydrate;
+
+    if (memoryCacheProvider.canFind<TModel>(query) && !requireRemote) {
       final memoryCacheResults = memoryCacheProvider.get<TModel>(query: query, repository: this);
 
+      if (alwaysHydrate) {
+        // start round trip for fresh data
+        // ignore: unawaited_futures
+        hydrate<TModel>(query: query, deserializeSqlite: !seedOnly);
```
Comment (mateominato) [inline packages/brick_offline_first/lib/src/offline_first_repository.dart:188]: do you prefer this lint ignore to wrapping in `unawaited`?

**Review by mateominato — APPROVED** [review]

File: packages/brick_offline_first/CHANGELOG.md (hunk 1 above)
Comment (tshedor) [reply packages/brick_offline_first/CHANGELOG.md:6]: Yeah for sure

File: packages/brick_offline_first/lib/src/offline_first_repository.dart (hunk 2 above)
Comment (tshedor) [reply packages/brick_offline_first/lib/src/offline_first_repository.dart:188]: Well...I genuinely don't know the difference. What would you favor?

Comment (mateominato) [reply packages/brick_offline_first/lib/src/offline_first_repository.dart:188]: there isn't; it's stylistic. but i think `unawaited` brings the decision into code and out of comments which i generally like

Comment (tshedor) [reply packages/brick_offline_first/lib/src/offline_first_repository.dart:188]: Ok yeah that's a great point

**Pull request merged**